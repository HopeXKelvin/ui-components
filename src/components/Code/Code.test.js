import React from 'react';
import { shallow, mount } from 'enzyme';

import { withTheme } from '../../utils/theme';

import { CopyNotice } from './Code';

import Code from '.';

document.execCommand = () => {};
document.getSelection = () => ({
  toString: () => '',
});

describe('<Code />', () => {
  let props;
  let wrapper;
  let onCopy;

  beforeEach(() => {
    onCopy = jest.fn();
    props = {
      onCopy,
    };
    wrapper = shallow(
      <Code {...props}>such code much programming</Code>,
      withTheme
    );
  });

  it('should call onCopy when the button is clicked', () => {
    wrapper = mount(
      withTheme(<Code {...props}>such code much programming</Code>)
    );
    expect(onCopy).not.toHaveBeenCalled();
    wrapper.find(CopyNotice).simulate('click');
    expect(onCopy).toHaveBeenCalledTimes(1);
  });

  it('should render correctly', () => {
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render multi line string correctly with multiCommand=true', () => {
    wrapper = shallow(
      <Code multiCommand>
        {`
sudo curl -l git.io/scope -o /usr/local/bin/scope
sudo chmod a+x /usr/local/bin/scope
scope launch https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
          `}
      </Code>,
      withTheme
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render multi line function correctly with multiCommand=true', () => {
    wrapper = shallow(
      <Code multiCommand>
        {() => `
sudo curl -l git.io/scope -o /usr/local/bin/scope
sudo chmod a+x /usr/local/bin/scope
scope launch https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
      `}
      </Code>,
      withTheme
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });

  it('should render multi line jsx correctly with multiCommand=true', () => {
    wrapper = shallow(
      <Code multiCommand>
        <div>sudo curl -L git.io/scope -o /usr/local/bin/scope</div>
        <div>sudo chmod a+x /usr/local/bin/scope</div>
        <div>
          scope launch <wbr />
          https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
        </div>
      </Code>,
      withTheme
    );
    expect(wrapper.dive()).toMatchSnapshot();
  });
});
