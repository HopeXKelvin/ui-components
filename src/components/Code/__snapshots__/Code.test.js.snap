// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Code /> should render correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Code__CodeWrapper-sc-1no9z1g-1",
        "isStatic": false,
        "lastClassName": "hZydFr",
        "rules": Array [
          "position:relative;background-color:",
          [Function],
          ";box-sizing:border-box;border-radius:",
          [Function],
          ";cursor:pointer;&:hover ",
          ".Code__CopyNotice-sc-1no9z1g-0",
          "{opacity:0.9;}",
        ],
      },
      "displayName": "Code__CodeWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Code__CodeWrapper-sc-1no9z1g-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Code__ScrollWrap>
    <Code__Content>
      <Code__Pre>
        such code much programming
      </Code__Pre>
    </Code__Content>
  </Code__ScrollWrap>
  <Code__CopyNotice
    isCopying={false}
    isHovered={false}
  >
    Copy to clipboard
  </Code__CopyNotice>
</StyledComponent>
`;

exports[`<Code /> should render multi line function correctly with multiCommand=true 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Code__CodeWrapper-sc-1no9z1g-1",
        "isStatic": false,
        "lastClassName": "hZydFr",
        "rules": Array [
          "position:relative;background-color:",
          [Function],
          ";box-sizing:border-box;border-radius:",
          [Function],
          ";cursor:pointer;&:hover ",
          ".Code__CopyNotice-sc-1no9z1g-0",
          "{opacity:0.9;}",
        ],
      },
      "displayName": "Code__CodeWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Code__CodeWrapper-sc-1no9z1g-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Code__ScrollWrap>
    <Code__Content>
      <Code__Pre>
        <Code__PaddedLine
          key="sudo curl -l git.io/scope -o /usr/local/bin/scope"
        >
          sudo curl -l git.io/scope -o /usr/local/bin/scope
          

        </Code__PaddedLine>
        <Code__PaddedLine
          key="sudo chmod a+x /usr/local/bin/scope"
        >
          sudo chmod a+x /usr/local/bin/scope
          

        </Code__PaddedLine>
        <Code__PaddedLine
          key="scope launch https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works"
        >
          scope launch https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
          

        </Code__PaddedLine>
      </Code__Pre>
    </Code__Content>
  </Code__ScrollWrap>
  <Code__CopyNotice
    isCopying={false}
    isHovered={false}
  >
    Copy to clipboard
  </Code__CopyNotice>
</StyledComponent>
`;

exports[`<Code /> should render multi line jsx correctly with multiCommand=true 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Code__CodeWrapper-sc-1no9z1g-1",
        "isStatic": false,
        "lastClassName": "hZydFr",
        "rules": Array [
          "position:relative;background-color:",
          [Function],
          ";box-sizing:border-box;border-radius:",
          [Function],
          ";cursor:pointer;&:hover ",
          ".Code__CopyNotice-sc-1no9z1g-0",
          "{opacity:0.9;}",
        ],
      },
      "displayName": "Code__CodeWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Code__CodeWrapper-sc-1no9z1g-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Code__ScrollWrap>
    <Code__Content>
      <Code__Pre>
        <Code__PaddedLine
          key="0/.0"
        >
          <div>
            sudo curl -L git.io/scope -o /usr/local/bin/scope
          </div>
          

        </Code__PaddedLine>
        <Code__PaddedLine
          key="1/.1"
        >
          <div>
            sudo chmod a+x /usr/local/bin/scope
          </div>
          

        </Code__PaddedLine>
        <Code__PaddedLine
          key="2/.2"
        >
          <div>
            scope launch 
            <wbr />
            https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
          </div>
          

        </Code__PaddedLine>
      </Code__Pre>
    </Code__Content>
  </Code__ScrollWrap>
  <Code__CopyNotice
    isCopying={false}
    isHovered={false}
  >
    Copy to clipboard
  </Code__CopyNotice>
</StyledComponent>
`;

exports[`<Code /> should render multi line string correctly with multiCommand=true 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Code__CodeWrapper-sc-1no9z1g-1",
        "isStatic": false,
        "lastClassName": "hZydFr",
        "rules": Array [
          "position:relative;background-color:",
          [Function],
          ";box-sizing:border-box;border-radius:",
          [Function],
          ";cursor:pointer;&:hover ",
          ".Code__CopyNotice-sc-1no9z1g-0",
          "{opacity:0.9;}",
        ],
      },
      "displayName": "Code__CodeWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Code__CodeWrapper-sc-1no9z1g-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  onClick={[Function]}
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  <Code__ScrollWrap>
    <Code__Content>
      <Code__Pre>
        <Code__PaddedLine
          key="sudo curl -l git.io/scope -o /usr/local/bin/scope"
        >
          sudo curl -l git.io/scope -o /usr/local/bin/scope
          

        </Code__PaddedLine>
        <Code__PaddedLine
          key="sudo chmod a+x /usr/local/bin/scope"
        >
          sudo chmod a+x /usr/local/bin/scope
          

        </Code__PaddedLine>
        <Code__PaddedLine
          key="scope launch https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works"
        >
          scope launch https://yjsjsubdx1h8un1f858gp7to8d51zdre@frontend.dev.weave.works
          

        </Code__PaddedLine>
      </Code__Pre>
    </Code__Content>
  </Code__ScrollWrap>
  <Code__CopyNotice
    isCopying={false}
    isHovered={false}
  >
    Copy to clipboard
  </Code__CopyNotice>
</StyledComponent>
`;
